web/client/reducers/config.js

```diff
@@ -69,7 +69,9 @@ function mapConfig(state = null, action) {
             }),
             mapConfigRawData: { ...action.config }
         };
-        newMapState.map = assign({}, newMapState.map, {mapId: action.mapId, size, version: hasVersion ? action.config.version : 1});
+        // if map is loaded from an already saved map keep the same id
+        const mapId = action.config?.fileName && state.map.present.mapId ? state.map.present.mapId : action.mapId;
```

Comment:
- [ ]  I do not think there be the case where  action.config?.fileName is falsy and state.map.present.mapId is truthy,
but the opposite should be possible when you import on a new map, so please test this use case

- [ ] and check also if it is already present the case where the mapId is taken from the action

- [ ] when you first load a map `action.config?.fileName` is undefined and `state.map` is null therefore it does not causes a problem, but i would make it more robust the second half of the condition by adding optional chain operator ?.  ` =>  state.map?.present?.mapId`

Reply:
> Added one test when an external config file is loaded and current map has an id

thank you